internal/embed/templates/peer.tmpl

```diff
@@ -85,8 +86,9 @@ protocol bgp {{ UniqueProtocolName $peer.ProtocolName $af }} {
 
             {{ if BoolDeref $peer.AllowBlackholeCommunity }}process_blackholes();{{ end }}
 
+            {{ if not $peer.DefaultBGPLocalPref }}
             bgp_local_pref = {{ $peer.LocalPref }}; # pathvector:localpref
```

Comment:
The optimizer config parser requires the `pathvector:localpref` annotation so this will break the optimizer. I'm not sure what the best way is to solve this; at a minimum the optimizer should expect certain configs to not have this annotation.

Reply:
I thought about it a bit more and I think the best solution here is to remove the outer if statement here and replace it with an inline one that sets `bgp_local_pref` to either `$peer.LocalPref` or `$peer.DefaultBGPLocalPref`

Reply:
Good point. Alternatively, we go with your original if statement and add a check in the config validator that ensures `DefaultLocalPref` and `OptimizeInbound` aren't both set.

Reply:
As said in Discord, I have no idea how to do that config validator thing. Also it seems my branch is quite out-of-date in the terms of the documentation layout for example.

Reply:
For the config validator, add an if statement to the end of the peer loop (https://github.com/natesales/pathvector/blob/main/internal/process/process.go#L230), logging a warning if `peerData.DefaultLocalPref && peerData.OptimizeInbound`

Reply:
You should also rebase against main to pull in the latest changes